src/Symfony/Component/Filesystem/Tests/PathTest.php

```diff
@@ -0,0 +1,1341 @@
+            array(false, '/webmozart/puli/style.CSS', 'css', false),
+            array(true, '/webmozart/puli/style.CSS', 'css', true),
+            array(true, '/webmozart/puli/style.ÄÖÜ', 'ÄÖÜ', false),
+
+            array(true, '/webmozart/puli/style.css', array('ext', 'css'), false),
+            array(true, '/webmozart/puli/style.css', array('.ext', '.css'), false),
+            array(true, '/webmozart/puli/style.css.', array('ext', ''), false),
+            array(false, '/webmozart/puli/style.css', array('foo', 'bar', ''), false),
+            array(false, '/webmozart/puli/style.css', array('.foo', '.bar', ''), false),
+        );
+
+        if (extension_loaded('mbstring')) {
+            // This can only be tested, when mbstring is installed
+            $tests[] = array(true, '/webmozart/puli/style.ÄÖÜ', 'äöü', true);
+            $tests[] = array(true, '/webmozart/puli/style.ÄÖÜ', array('äöü'), true);
+        }
+
+        return $tests;
+    }
+
+    /**
+     * @dataProvider provideHasExtensionTests
+     */
+    public function testHasExtension($hasExtension, $path, $extension, $ignoreCase)
+    {
+        $this->assertSame($hasExtension, Path::hasExtension($path, $extension, $ignoreCase));
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The path must be a string. Got: array
+     */
+    public function testHasExtensionFailsIfInvalidPath()
+    {
+        Path::hasExtension(array());
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The extensions must be strings. Got: stdClass
+     */
+    public function testHasExtensionFailsIfInvalidExtension()
+    {
+        Path::hasExtension('/style.css', (object)array());
+    }
+
+    public function provideChangeExtensionTests()
+    {
+        return array(
+            array('/webmozart/puli/style.css.twig', 'html', '/webmozart/puli/style.css.html'),
+            array('/webmozart/puli/style.css', 'sass', '/webmozart/puli/style.sass'),
+            array('/webmozart/puli/style.css', '.sass', '/webmozart/puli/style.sass'),
+            array('/webmozart/puli/style.css', '', '/webmozart/puli/style.'),
+            array('/webmozart/puli/style.css.', 'twig', '/webmozart/puli/style.css.twig'),
+            array('/webmozart/puli/style.css.', '', '/webmozart/puli/style.css.'),
+            array('/webmozart/puli/style.css', 'äöü', '/webmozart/puli/style.äöü'),
+            array('/webmozart/puli/style.äöü', 'css', '/webmozart/puli/style.css'),
+            array('/webmozart/puli/', 'css', '/webmozart/puli/'),
+            array('/webmozart/puli', 'css', '/webmozart/puli.css'),
+            array('/', 'css', '/'),
+            array('', 'css', ''),
+        );
+    }
+
+    /**
+     * @dataProvider provideChangeExtensionTests
+     */
+    public function testChangeExtension($path, $extension, $pathExpected)
+    {
+        static $call = 0;
```

Comment:
What is the purpose of that variable?

Reply:
No idea, but it doesn't seem to be used so I removed it